soda_machine.rb

```diff
@@ -7,12 +9,26 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.count
   end
 
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      return soda if soda.brand == soda_brand
+    end
+      return nil
```

Comment:
Nice.  This last line should be unindented.  Also, as we discussed in the code-through, you can just use the built-in `.find` for this.  Basically, if there's something you want to do to manipulate and array or other enumerable, look in the docs first.  It's probably in there :)